modules/authentication/src/models/TwoFactorBackUpCodes.schema.ts

```diff
@@ -0,0 +1,48 @@
+import { ConduitActiveSchema, DatabaseProvider, TYPE } from '@conduitplatform/grpc-sdk';
+import { User } from './User.schema';
+
+const schema = {
+  _id: TYPE.ObjectId,
+  user: {
+    type: TYPE.Relation,
+    model: 'User',
+    required: true,
+  },
+  codes: [TYPE.String],
+  createdAt: TYPE.Date,
+  updatedAt: TYPE.Date,
+};
+
+const modelOptions = {
+  timestamps: true,
+  conduit: {
+    permissions: {
+      extendable: true,
+      canCreate: false,
+      canModify: 'ExtensionOnly',
+      canDelete: false,
+    },
+  },
+} as const;
+const collectionName = undefined;
+
+export class TwoFactorBackUpCodes extends ConduitActiveSchema<TwoFactorBackUpCodes> {
+  private static _instance: TwoFactorBackUpCodes;
+  _id: string;
+  user: string | User;
+  codes: string[];
```

Comment:
Why generate multiple codes in the first place?
Is that merely so we can invalidate used ones upon recovery without worrying about the user not receiving/recreating new ones?

Regardless, why not use a 1:N relation instead of storing these in a single array?